wcfsetup/install/files/lib/system/database/Database.class.php

```diff
@@ -332,6 +334,33 @@ public function prepareStatement($statement, $limit = 0, $offset = 0)
         }
     }
 
+    /**
+     * Prepares a statement for execution and returns a statement object.
+     *
+     * In contrast to `prepareStatement()`, for all installed apps, `app1_` is replaced with
+     * `app{WCF_N}_`.
+     */
+    public function prepare(string $statement, int $limit = 0, int $offset = 0): PreparedStatement
+    {
+        static $abbreviations = null;
+        if ($abbreviations === null) {
+            $abbreviations = \implode(
+                '|',
+                \array_map(static function (Application $app): string {
+                    return $app->getAbbreviation();
+                }, ApplicationHandler::getInstance()->getApplications())
+            );
+        }
+
+        $statement = \preg_replace(
+            "~(\b(?:{$abbreviations}))1_~",
+            '${1}' . WCF_N . '_',
```

Comment:
```suggestion
            '\\1' . WCF_N . '_',
```
The `${1}` syntax is super strange. I've never seen this in use.

Reply:
With `WCF_N = 1`, `'\\1' . WCF_N . '_'` becomes `'\\11_',` so that `${1}` has to be used.

Reply:
Ah, I see.